contracts/contracts/system_contracts/kaiabridge/IGuardian.sol

```diff
@@ -0,0 +1,171 @@
+    /// @param _minGuardianRequiredConfirm Number of required confirmations.
+    function changeRequirement(uint8 _minGuardianRequiredConfirm) external virtual;
+
+    /// @dev Allows an guardian to submit and confirm a transaction.
+    /// @param to target contract address
+    /// @param data Transaction data payload.
+    /// @param uniqUserTxIndex unique user input number to be mapped with tx ID
+    function submitTransaction(address to, bytes calldata data, uint256 uniqUserTxIndex)
+        external
+        virtual
+        returns (uint256);
+
+    /// @dev Allows an guardian to confirm a transaction.
+    /// @param txID Transaction ID.
+    function confirmTransaction(uint256 txID) external virtual;
+
+    /// @dev Allows an guardian to revoke a confirmation for a transaction.
+    /// @param txID Transaction ID.
+    function revokeConfirmation(uint256 txID) external virtual;
+
+    /// @dev Allows anyone to execute a confirmed transaction.
+    /// @param txID Transaction ID.
+    function executeTransaction(uint256 txID) external virtual;
+
+    /// @dev Returns the confirmation status of a transaction.
+    /// @param txID Transaction ID.
+    /// @return Confirmation status.
+    function isConfirmed(uint256 txID) external virtual view returns (bool);
+
+    /// @dev Returns number of confirmations of a transaction.
+    /// @param txID Transaction ID.
+    /// @return Number of confirmations.
+    function getConfirmationCount(uint256 txID) external virtual view returns (uint256);
+
+    /// @dev Returns total number of transactions after filers are applied.
+    /// @param pending Include pending transactions.
+    /// @param executed Include executed transactions.
+    /// @return Total number of transactions after filters are applied.
+    function getTransactionCount(bool pending, bool executed) external virtual view returns (uint256, uint256);
+
+    /// @dev Returns list of guardians.
+    /// @return List of guardian addresses.
+    function getGuardians() external virtual view returns (address[] memory);
+
+    /// @dev Returns array with guardian addresses, which confirmed transaction.
+    /// @param txID Transaction ID.
+    /// @return Returns array of guardian addresses.
+    function getConfirmations(uint256 txID) external virtual view returns (address[] memory);
+
+    /// @dev Returns list of transaction IDs in defined range.
+    /// @param from Index start position of transaction array.
+    /// @param to Index end position of transaction array.
+    /// @param pending Include pending transactions.
+    /// @param executed Include executed transactions.
+    /// @return Returns array of transaction IDs.
+    function getTransactionIds(uint256 from, uint256 to, bool pending, bool executed)
+        external
+        virtual
+        view
+        returns (uint256[] memory, uint256[] memory);
+
+
+    //////////////////// Storage variables ////////////////////
+    address[] public guardians;
```

Comment:
Why don't we use `EnumerableSet.AddressSet` for `address[]` type? If they're `set` and need to be added/removed, then using `EnumerableSet` is more efficient.

Reply:
I thought the primitive array seems more intuitive and simple and seems better suited to covering the operations related to those types.

Reply:
Yes, but I can't find operations better suited to array. 
For example, adding/removing elems from arrays is more efficient in EnumerableSet.

Could you please give me a simple example? Thanks.

Reply:
Yes, I agree that it's more intuitive, but I think EnumerableSet would be more efficient for operations it needed. (e.g., adding/removing elems)

If it's a little tight to change the implementation now, I think it's fine as is, thank you.